### PR TITLE
Chore: More cosmetics

### PR DESCRIPTION
**Pull request by amotl**
Hi there,

this brings in more cosmetics, mostly related to CI.

- Use `crate/crate:nightly` for running tests.
- Explicitly use `ubuntu-18.04` because the upcoming `ubuntu-20.04` will only support `PHP>=7.4`.
- Add PHP code style checks.
- Upgrade docs build system, see #87.
- Also build documentation on GHA.

With kind regards,
Andreas.

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,14 +1,5 @@
-# don't pin crate version numbers so the latest will always be pulled when you
-# set up your environment from scratch
+setuptools>=41.1.0
+sphinx>=1.8.5,<4
+sphinx-autobuild
```
Comment (mechanomi) [inline docs/requirements.txt:3]: this should just be `crate-docs-theme`. the rest is handled by the Makefile or RTD, respectively

see https://github.com/crate/crate-docs/blob/main/docs/requirements.txt

unless there is a good reason to deviate, https://github.com/crate/crate-docs/blob/main/docs should be considered the template project for the common build :)

File: DEVELOP.rst [hunk 2]
```diff
@@ -33,7 +39,55 @@ You can run the tests like so::
...
+
+    # Run code style checks
+    composer run style
+
+    # Run code style checks
+    ./vendor/bin/phpcs ./src
+
+
+*************
+Documentation
```
Comment (mechanomi) [inline DEVELOP.rst:89]: can you add a "Working on the documentation" section? https://github.com/crate/crate-docs/blob/main/DEVELOP.rst is the template, so you can copy and paste from there

**Review by mechanomi — CHANGES_REQUESTED** [review]
nice!! another common build docs project! :D

File: docs/requirements.txt (hunk 1 above)
Comment (amotl) [reply docs/requirements.txt]: Thanks. Addressed it.

File: DEVELOP.rst (hunk 2 above)
Comment (amotl) [reply DEVELOP.rst]: Improved the title, thanks.

Comment (amotl) [reply DEVELOP.rst]: Ah, do you think I should copy in the whole section?

Comment (amotl) [reply DEVELOP.rst]: Done.

File: DEVELOP.rst [hunk 3]
```diff
@@ -29,11 +35,105 @@ set up your remote interpreter and test environment.
...
+Installation
+============
+
+Clone the project::
+
+    git clone git@github.com:crate/crate-dbal.git
+
+Setup environment::
+
+    composer install --prefer-source
```
Comment (seut) [inline DEVELOP.rst:67]: does this require to install `composer` + `php` + dependend libraries on the host maschine? I would not recommend doing so... I really would not want my local host system polluted with this. I think this should document how to run a 2nd php container with workspace mounted.

File: DEVELOP.rst [hunk 4]
```diff
@@ -29,11 +35,105 @@ set up your remote interpreter and test environment.
...
+      reset   Reset the build
+
+You must install `fswatch`_ to use the ``dev`` target.
+
+
+Continuous integration and deployment
+=====================================
+
+CI is `configured`_ to run ``make check`` from the ``docs`` directory.
+Please do not merge pull requests until the tests pass.
```
Comment (seut) [inline DEVELOP.rst:129]: I'd really be in favor of setting this up as a strict rule enforced by the CI and not a loose advise. Isn't it already enforced?

Comment (mechanomi) [reply DEVELOP.rst:129]: this has been set up as a strict rule on all repositories. this advice is technically redundant now. I would be okay with removing the second sentence

we should remove it from all docs projects while we're at it

what do you think, @amotl?

**Review by mechanomi — APPROVED** [review]
looks good. but I agree with @seut on removing that sentence

File: DEVELOP.rst (hunk 4 above)
Comment (amotl) [reply DEVELOP.rst]: Sure thing. I've addressed your suggestion and also propagated that change to https://github.com/crate/crate-docs/pull/62. I believe there is no need to explicitly step through all repositories now, though.

**Comment by amotl** [comment]
Hi again,

two more things have been addressed.

- Adjust to modern Scrutinizer configuration.
- Adjust DEVELOP.rst title formatting.

With kind regards,
Andreas.

**Review by seut — APPROVED** [review]
👍

**Pull request merged**